packages/amplify-provider-awscloudformation/src/override-manager/transform-resource.ts

```diff
@@ -19,7 +19,8 @@ export async function transformResourceWithOverrides(context: $TSContext, resour
 
   try {
     if (resource) {
-      const { transformCategoryStack } = await import(`@aws-amplify/amplify-category-${resource.category}`);
+      const pluginInfo = context.amplify.getCategoryPluginInfo(context, resource.category, resource.resourceName);
+      const { transformCategoryStack } = pluginInfo ? await import(pluginInfo.packageLocation) : { transformCategoryStack: null };
```

Comment:
If this PR was reduced to these lines here, I think that would achieve the majority of use cases for https://github.com/aws-amplify/amplify-cli/issues/9226. Thoughts @fossamagna?

Reply:
Hi, @jhockett
> If this PR was reduced to these lines here, I think that would achieve the majority of use cases for https://github.com/aws-amplify/amplify-cli/issues/9226.

Yes, I think so.


> What is the use case for the changes made to auth, storage, and api?

I added `context` object to use `CLIInputSchemaValidator` on 3rd-party plugin at following.

https://github.com/aws-amplify/amplify-cli/pull/9601/files#diff-0261ff0abc6d6facc760fc5319008e2ebe0993de83bd0a2898a113c526a16372R149

https://github.com/fossamagna/amplify-category-console-notification/commit/961262f698294efa8651cfb893497ed537176975#diff-be6013127ca9dce2788dc277cc0fefa77620c5aee6a7ec82fcb41462b415c6d4R42

I think that it will be resolved by implementing equivalent `CLIInputSchemaValidator` in 3rd-party plugin.

Reply:
I understand now, thanks for the examples! I think we should include the plugin support for the schema generator too.

Reply:
@jhockett 
I think that it is unnecessary modify to support for the schema generator, because generated schema path resolved by relative path. If I overlooking what should be support for the schema generator, I would like to seek your advice on this.

https://github.com/aws-amplify/amplify-cli/blob/ae7b001f274f327a29c99c67fe851272c6208e84/packages/amplify-cli-core/src/category-interfaces/category-base-schema-generator.ts#L32-L33

Reply:
@fossamagna The schema generator takes a category's TS type for the input state and returns a JSON schema that the category's `cli-inputs.json` should conform to.

Example input state schema for REST APIs: https://github.com/aws-amplify/amplify-cli/blob/ae7b001f274f327a29c99c67fe851272c6208e84/packages/amplify-category-api/resources/schemas/aPIGateway/APIGatewayCLIInputs.schema.json

Where the schema validation logic is invoked: https://github.com/aws-amplify/amplify-cli/blob/ae7b001f274f327a29c99c67fe851272c6208e84/packages/amplify-category-api/src/provider-utils/awscloudformation/apigw-input-state.ts#L217

Reply:
@jhockett I understand now, schema generator is pointing to `CLIInputSchemaValidator` class in `category-base-schema-genereator.ts`, not `CLIInputSchemaGenerator` class.
Then, I don't think I have any part of this PR that needs to be corrected.
What do you think on this?

Reply:
Hey @fossamagna, sorry for the delayed response. Thinking about it more, this approach LGTM 👍